### ISSUE TITLE
Nop did not pass originating scheme (HTTP/HTTPS) to middlewares from proxy. 

### ISSUE THREAD

**Issue by nextfool** [nopSolutions/nopCommerce#5584]
nopCommerce version: all

If we deploy Nop behind a proxy, the HTTPS schema is not passed to those middlewares who may need the info (like the authentication middleware), this will cause the facebook auth plugin to send only HTTP callback url to facebook api.

To fix this, we need to add XForwardedProto header to request pipline.

https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0

```
        public void ConfigureRequestPipeline(IApplicationBuilder application)
        {
            application.UseForwardedHeaders(new ForwardedHeadersOptions
            {
                ForwardedHeaders = ForwardedHeaders.XForwardedProto
            });
            ServiceProvider = application.ApplicationServices;

            //find startup configurations provided by other assemblies
            var typeFinder = Resolve<ITypeFinder>();
            var startupConfigurations = typeFinder.FindClassesOfType<INopStartup>();

            //create and sort instances of startup configurations
            var instances = startupConfigurations
                .Select(startup => (INopStartup)Activator.CreateInstance(startup))
                .OrderBy(startup => startup.Order);

            //configure request pipeline
            foreach (var instance in instances)
                instance.Configure(application);
        }
```

**Comment by RomanovM** [comment]
This should work properly when no proxy is used.
See also discussion here https://www.nopcommerce.com/boards/topic/82570/facebook-login-not-working-in-43

**Comment by exileDev** [comment]
Closed #5584
